pkg/istio-agent/README.md

```diff
@@ -0,0 +1,82 @@
+the ca client to use certificates written to the same directory we have configured them to be written to.
+
+## Authentication
+
+The agent supports two forms of authentication with the CA/discovery servers: mTLS and JWT. Varying deployment
+topologies mix and match these two.
+
+For a standard Kubernetes deployment, both CA and discovery will use JWT authentication, with a token automatically
+[generated and rotated by Kubernetes](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#service-account-token-volume-projection).
+
+For discovery, the JWT token will be read directly from a file and sent as is. For CA, this logic is a bit more complex,
+as the support for external CAs is more mature than external discovery servers. This supports some additional
+configuration, a `CredentialFetcher` which allows fetching a token from places other than a file (for example, a local
+metadata server), and a `TokenExchanger` which allows exchanging a token for another form to match the CA server requirements.
+
+For VMs, the standard flow is for the user to provision a short-lived JWT token onto the VM. After the initial
+CSR, certificates are written to disk and mTLS is used for future requests. If the VM restarted, it would continue
+to use the certificates written to disk, assuming the downtime is less than certificate expiration. This is why
+the certificates are persisted to disk, rather than kept in memory like in the standard Kubernetes deployment.
+
+## Certificate Rotation
+
+The agent also handles rotating certificates near expiration. It does so by triggering a callback from the `SecretManager` to the SDS server
+when a certificate is near expiration (configurable by `SECRET_GRACE_PERIOD_RATIO`, defaulting to half of the expiration). If the SDS server
+is still interested in this certificate (ie, Envoy is still connected and requesting the certificate), the SDS server will send another request
+to generate a new secret and push the updated certificate to Envoy. This ensures that we do not permanently watch certificates even after
+Envoy has stopped requested them; if there are no subscriptions they update will be ignored. If Envoy later watches these certificates again,
+a new one will be generated on demand.
+
+## Configuration
+
+| Variable | Description |
+| - | - |
+|CA_ADDR|Address of CA, defaults to discoveryAddress|
+|CA_PROVIDER|Type of CA; supported values are GoogleCA or Citadel (although anything but GoogleCA will use Citadel); defaults to Citadel|
+|PROV_CERT|certificates to be used for mTLS communication with control plane only; NOT for workload mTLS|
+|OUTPUT_CERT|write all fetched certificates to some directory. Used to support applications that need certificates (Prometheus) as well as rotating mTLS control plane authentication.|
+|FILE_MOUNTED_CERTS|completely disable CA path, exclusively use certs mounted into the pod with set certificate file locations|
```

Comment:
Might be worth clarifying these two points a further - 
(A) if PROV_CERT IS provisioned, then will this cert always be used to authenticate with control-plane servers (irrespective of whatever certs we get after CSR?)? Also, is there a possibility that the PROV_CERT is different for reaching the SDS server vs the XDS server?
(B)Can PROV_CERT and FILE_MOUNTED_CERTS be provisioned at the same time? My understanding is no. If the user provisions FILE_MOUNTED_CERTS, then these certs are meant to be used for all kinds of communication - for data-plane and control-plane communication

Reply:
A) is why I was suggesting we have the same config schema for xds and CA. Currently its somewhat arbitrary which fields apply to which server. I will document this discrepency